tool/templates/cpp/runtime.hpp.jinja

```diff
@@ -228,6 +228,68 @@ private:
 
 #endif // __cplusplus >= 202002L
 
+// An ABI stable std::basic_string_view equivalent for the case of string
+// views in slices
+template <class CharT, class Traits = std::char_traits<CharT>>
+class basic_string_view_for_slice {
+public:
+  using std_string_view           = std::basic_string_view<CharT, Traits>;
+  using traits_type               = typename std_string_view::traits_type;
+  using value_type                = typename std_string_view::value_type;
+  using pointer                   = typename std_string_view::pointer;
+  using const_pointer             = typename std_string_view::const_pointer;
+  using size_type                 = typename std_string_view::size_type;
+  using difference_type           = typename std_string_view::difference_type;
+
+  constexpr basic_string_view_for_slice() noexcept
+    : basic_string_view_for_slice{std_string_view{}} {}
+
+  constexpr basic_string_view_for_slice(const basic_string_view_for_slice& other) noexcept = default;
+
+  constexpr basic_string_view_for_slice(const const_pointer s, const size_type count)
+    : basic_string_view_for_slice{std_string_view{s, count}} {}
+
+  constexpr basic_string_view_for_slice(const const_pointer s)
+    : basic_string_view_for_slice{std_string_view{s}} {}
+
+  constexpr basic_string_view_for_slice& operator=(const basic_string_view_for_slice& view) noexcept = default;
+
+  constexpr basic_string_view_for_slice(const std_string_view& s) noexcept
+    : data_{s.data(), s.size()} {}
+
+  constexpr basic_string_view_for_slice& operator=(const std_string_view& s) noexcept {
+    data_ = {s.data(), s.size()};
+    return *this;
+  }
+
+  constexpr operator std_string_view() const noexcept { return {data(), size()}; }
+  constexpr std_string_view as_sv() const noexcept { return *this; }
+
+  constexpr const_pointer data() const noexcept { return data_.data; }
+  constexpr size_type size() const noexcept { return data_.len; }
+
+private:
+  using capi_type =
+    std::conditional_t<std::is_same_v<value_type, char>,
```

Comment:
shouldn't this be `unsigned char` aka `char8_t`?

Reply:
And yeah this has been an annoyance in the past, C++ uses `char` not `char8_t` here.